app/middlewares/validatonSchemas/users.js

```diff
@@ -0,0 +1,54 @@
+const { user: UserModel } = require('../../models/index');
+
+const UserSchemas = module.exports;
+
+const providedAndNotEmpty = 'should be provided and not empty';
+
+UserSchemas.createUserSchema = {
+  first_name: {
+    in: 'body',
+    optional: false,
+    notEmpty: true,
+    errorMessage: providedAndNotEmpty
+  },
+  last_name: {
+    in: 'body',
+    optional: false,
+    notEmpty: true,
+    errorMessage: providedAndNotEmpty
+  },
+  email: {
+    in: 'body',
+    optional: false,
+    notEmpty: true,
+    errorMessage: providedAndNotEmpty,
+    isEmail: {
+      errorMessage: 'should be a valid email',
+      bail: true
+    },
+    matches: {
+      options: /(\w|\.|-)+@wolox\.(\w{2}|com)$/i,
+      errorMessage: 'should belong to wolox domain',
+      bail: true
+    },
+    custom: {
+      options: email =>
+        UserModel.findOne({ where: { email } }).then(user => {
```

Comment:
Nice!

Reply:
What happen if the db is down ?

Reply:
if the database is down, the service would throw a SequelizeConnectionError when we try to start the server, if the database is up but there is a database error in this validation we would throw a badRequestError with message "email database error".